### PR TITLE
Adds experimental event component responder surfaces

### PR DESCRIPTION
**Pull request by trueadm**
Note: this is for an experimental event API that we're testing out internally at Facebook.

This PR adds Press and Hover responder surfaces, some additional event context APIs and fixes some bugs surfaced from the additional Press tests added.

#15036
#15112
#15150
#15168
#15179

This PR also adds tests that validate the triggering of event responder handleEvent function.

### CODE REVIEW

File: packages/react-events/src/Press.js [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+  },
+  handleEvent(
+    context: EventResponderContext,
+    props: Object,
+    state: PressState,
+  ): void {
+    const {eventTarget, eventType, event} = context;
+
+    switch (eventType) {
+      case 'keydown': {
```
Comment (necolas) [inline packages/react-events/src/Press.js]: I think we should consider making keyboards trigger `pressin`, `pressout`, `press`, and even `longpress` events.

Comment (trueadm) [reply packages/react-events/src/Press.js]: Will do this at a later point, in a follow up.

**Comment by trueadm** [comment]
I checked manually and this PR does not add any size compared to master. The results.json in master needs updating.

**Comment by sebmarkbage** [comment]
We don't compare against the results.json file but a rebuilt version. I tested locally and the (small) build increase shown by sizebot above seems legit.

**Comment by trueadm** [comment]
@sebmarkbage I saw it was approved earlier, we also spoke about it earlier too. None of these code paths should get used on FB, but I’ll be sure to sanity check this.

 The size difference doesn’t make sense. If I remove the changes in this PR and build, I still get 0.3% increase locally.

**Comment by sebmarkbage** [comment]
I checked out this commit built and then checked out the previous commit and built that. Then compared the two bundles.

I never compare against the checked in JSON. That one never matches up.

**Pull request merged**